netcore/NetCore.Session/NetCoreSessionState.cs

```diff
@@ -0,0 +1,299 @@
+        return null;
+      }
+
+      // Check for stale key
+      if (ssValue.SpanUntilStale != TimeSpan.Zero &&
+          nowUtc > (ssValue.LastAccessTimeUtc + ssValue.SpanUntilStale)) {
+        return null;
+      }
+
+      // Update the times for sliding expirations
+      if (ssValue.SpanUntilStale != TimeSpan.Zero) {
+        ssValue.LastAccessTimeUtc = nowUtc;
+        _region.PutByteArray(key, ssValue.ToByteArray());
+      }
+
+      return ssValue.Value;
+    }
+
+    public Task<byte[]> GetAsync(string key, CancellationToken token = default(CancellationToken)) {
+      if (key == null) {
+        throw new ArgumentNullException(nameof(key));
+      }
+
+      token.ThrowIfCancellationRequested();
+
+      return Task.Factory.StartNew(() => Get(key), token);
+    }
+
+    public void Refresh(string key) {
+      if (key == null) {
+        throw new ArgumentNullException(nameof(key));
+      }
+
+      Connect();
+
+      // Check for nonexistent key
+      GeodeSessionStateValue ssValue = GetValueForKey(key);
+      if (ssValue == null) {
+        return;
+      }
+
+      // Check for expired key
+      var nowUtc = DateTime.UtcNow;
+      if (ssValue.ExpirationTimeUtc != DateTime.MinValue && ssValue.ExpirationTimeUtc < nowUtc) {
+        return;
+      }
+
+      // Check for stale key
+      if (ssValue.SpanUntilStale != TimeSpan.Zero &&
+          nowUtc > (ssValue.LastAccessTimeUtc + ssValue.SpanUntilStale)) {
+        return;
+      }
+
+      // Update the times for sliding expirations
+      if (ssValue.SpanUntilStale != TimeSpan.Zero) {
+        ssValue.LastAccessTimeUtc = nowUtc;
+        _region.PutByteArray(key, ssValue.ToByteArray());
+      }
+    }
+
+    public Task RefreshAsync(string key, CancellationToken token = default(CancellationToken)) {
+      if (key == null) {
+        throw new ArgumentNullException(nameof(key));
+      }
+
+      token.ThrowIfCancellationRequested();
+
+      return Task.Factory.StartNew(() => Refresh(key), token);
+    }
+
+    public void Remove(string key) {
+      if (key == null) {
+        throw new ArgumentNullException(nameof(key));
+      }
+
+      Connect();
+
+      _region.Remove(key);
+    }
+
+    public Task RemoveAsync(string key, CancellationToken token = default(CancellationToken)) {
+      if (key == null) {
+        throw new ArgumentNullException(nameof(key));
+      }
+
+      token.ThrowIfCancellationRequested();
+
+      return Task.Factory.StartNew(() => Remove(key), token);
+    }
+
+    public void Set(string key, byte[] value, DistributedCacheEntryOptions options) {
+      if (key == null) {
+        throw new ArgumentNullException(nameof(key));
+      }
+
+      if (value == null) {
+        throw new ArgumentNullException(nameof(value));
+      }
+
+      if (options == null) {
+        throw new ArgumentNullException(nameof(options));
+      }
+
+      Connect();
+
+      GeodeSessionStateValue ssValue = new GeodeSessionStateValue();
```

Comment:
`var` seems to have been missed in many places in this file.

Reply:
Good catch. Too many years of no var, no auto.